src/Symfony/Component/Translation/Translator.php

```diff
@@ -189,10 +219,14 @@ public function setFallbackLocale($locales)
      *
      * @throws \InvalidArgumentException If a locale contains invalid characters
      *
+     * @deprecated since version 2.8, to be removed in 3.0. Use FallbackCatalogueBag::setFallbackLocales instead.
+     *
      * @api
      */
     public function setFallbackLocales(array $locales)
```

Comment:
Haven't run your code, but further down you possibly use the CatalogueBag to obtain the catalogues, but you do not pass on the fallback locales?